docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,501 @@
+- Doesn’t provide separation of responsibilities between creating the general cloud infrastructure for the cluster and the actual cluster control plane.
+- Managed Kubernetes look different from unmanaged Kubernetes where two separate kinds are used for a control plane and infrastructure. This would impact products building on top of CAPI.
+
+#### Option 2: Two kinds with a ControlPlane and a pass-through InfraCluster
+
+This option introduces 2 new resource kinds:
+
+- **GCPManagedControlPlane**: same as in option 1
+- **GCPManagedCluster**: contains the minimum properties in its spec and status to satisfy the [CAPI contract for an infrastructure cluster](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html) (i.e. ControlPlaneEndpoint, Ready condition). Its controller watches GCPManagedControlPlane and copies the ControlPlaneEndpoint field to GCPManagedCluster to report back to CAPI. This is used as a pass-through layer only.
+
+```go
+type GCPManagedControlPlaneSpec struct {
+// Project is the name of the project to deploy the cluster to.
+Project string `json:"project"`
+
+// NetworkSpec encapsulates all things related to the GCP network.
+// +optional
+Network NetworkSpec `json:"network"`
+
+// AddonsConfig defines the addons to enable with the GKE cluster.
+// +optional
+AddonsConfig *AddonsConfig `json:"addonsConfig,omitempty"`
+
+// Logging contains the logging configuration for the GKE cluster.
+// +optional
+Logging *ControlPlaneLoggingSpec `json:"logging,omitempty"`
+
+// EnableKubernetesAlpha will indicate the kubernetes alpha features are enabled
+// +optional
+EnableKubernetesAlpha bool
+
+// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+// +optional
+ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+....
+}
+```
+
+```go
+type GCPManagedClusterSpec struct {
+// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+// +optional
+ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+}
+```
+
+**This is the design pattern used by AKS in CAPZ**. [An example of how ManagedCluster watches ControlPlane in CAPZ.](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/controllers/azuremanagedcluster_controller.go#L165)
+
+**Pros**
+
+- Better aligned with CAPI’s traditional infra provider model
+- Works with ClusterClass
+
+**Cons**
+
+- Need to maintain Infra cluster kind, which is a pass-through layer and has no other functions. In addition to the CRD, controllers, webhooks and conversions webhooks need to be maintained.
+- Infra provider doesn’t provision infrastructure and whilst it may meet the CAPI contract, it doesn’t actually create infrastructure as this is done via the control plane.
+
+#### Option 3: Two kinds with a Managed Control Plane and Managed Infra Cluster with Better Separation of Responsibilities
```

Comment:
How will clusterctl get kubeconfig supposed to work in these scenarios? For example in case of OKE, Auth tokens are used as explained here  - https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengaccessingclusterkubectl.htm
How will core capi components access the workload cluster apiserver in such scenario?

Reply:
Resolving the comment as it is answered in [slack](https://kubernetes.slack.com/archives/CD6U2V71N/p1659952785791429)